nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/NuxeoCloudClient.java

```diff
@@ -330,13 +336,34 @@ public String uploadDataset(@Nonnull DocumentModel dataset) {
             try {
                 DateTime start = DateTime.now();
                 BatchUpload batchUpload = client.getBatchUpload(CHUNK_100_MB);
-
                 String batch1 = batchUpload.getBatchId();
-                batch1 = createBatch(batchUpload, "training", "0", trainingData) ? batch1 : null;
+                if (trainingData == null) {
+                    log.warn("Job/Command: {} has no training data - null data", jobId);
+                    batch1 = null;
+                } else if (trainingData.getLength() == 0) {
+                    log.warn("Job/Command: {} has no training data - data has 0 length", jobId);
+                    batch1 = null;
+                } else if (trainingData.getFile() == null) {
+                    log.warn("Job/Command: {} has no training data - data blob file is null", jobId);
+                    batch1 = null;
+                } else {
+                    batch1 = createBatch(batchUpload, "training", "0", trainingData) ? batch1 : null;
+                }
 
```

Comment:
Maybe just to have smaller methods, for better readability, we can extract methods from lines 338 to 351 and from 353 to 366 (including the lines with the assignments to `batchUpload` variable like this `batchUpload = client.getBatchUpload(CHUNK_100_MB);`)

And set the names for methods (for example): `createTrainingDataBatch()`, `createEvalDataBatch()`

Reply:
Thank you, @nichiporenko! I think that's a good idea. Pushing a change for this now.

Reply:
This has been addressed with 6350432 - https://github.com/nuxeo/nuxeo-ai/compare/7f01bfb1280f48e707a1ad6e538842fe508cc602..63504328f34dade871510773dd4e9c18e567a456

Reply:
Looks good! Thanks for the update